### PR TITLE
Implement Linear context provider 

### PR DESCRIPTION
**Pull request by valerybugakov**
- Implements Linear context provider for OpenCtx, which allows to search for issues in Linear and use an issue title, description, and comments as AI content.
- Uses [@linear/sdk](https://developers.linear.app/docs/graphql/working-with-the-graphql-api#linear-sdk) to fetch issues
- Requires authorization in a Linear OAuth2 application or a personal API token

## Test plan

- Tested using [an OAuth2 application](https://linear.app/sourcegraph/settings/api/applications/a5c42ede-7768-45e7-bc39-e32b3208ceaf/edit) created in the Sourcegraph workspace

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
     "**/dist": true
   },
   "files.exclude": {
-    "**/dist": true,
```
Comment (keegancsmith) [inline .vscode/settings.json:7]: @valerybugakov out of interest what is the purpose of this change?

Comment (valerybugakov) [reply .vscode/settings.json:7]: I should have tagged you or other folks on that. It allows me to inspect built artifacts using the file tree sidebar. We bumped into some issues related to how providers are bundled and then dynamically loaded in the VS Code extension. 

I can revert if it messes up your local setup.

Comment (valerybugakov) [reply .vscode/settings.json:7]: Gotcha!
